### ISSUE TITLE
(Tutorials) documentation revamp

### ISSUE THREAD

**Issue by xlxs4** [poliastro/poliastro#1124]
💡 **Feature request**

Hello, the [Tutorials](https://docs.poliastro.space/en/latest/gallery.html) documentation is a great idea to ease in newcomers to the library! As I was working through the examples, I noticed that, while the texts are well-written and to the point, there's some (maybe inherent/inevitable) inconsistency between different tutorials, as well as small (mostly grammatical) errors here and there.

The inconsistencies mentioned can range from very little nitpicks that probably only bother me (e.g. some sentences before code blocks end with a `.`, others with a `:`; poliastro is sometimes written _poliastro_; `from astropy import units as u` vs `import astropy.units as u`...) to more major differences between tutorials (check [this](https://docs.poliastro.space/en/latest/examples/Going%20to%20Mars%20with%20Python%20using%20poliastro.html) and compare it to [this](https://docs.poliastro.space/en/latest/examples/Comparing%20Hohmann%20and%20bielliptic%20transfers.html)).

Additionally, some extra work could be put into this, for example as per https://github.com/poliastro/poliastro/issues/949. 
An objection to this, however could very well be that:
a) the text is in a pretty good state as it is, and
b) a technical writer will probably get to work on this sooner or later (https://github.com/poliastro/poliastro/issues/1106)

Nonetheless, I believe small, sporadic changes can happen to polish some text sections without requiring significant effort. I could also try to make some iterations, if I can. I intend to hit up a small PR addressing only minor grammatical fixes in a little bit.

🎯 **Goal**

As someone who heavily focuses on the presentation of an idea/project, I think small enhancements such as these mentioned above could take the already great impression `poliastro` leaves on a newcomer and up it just a notch.

More importantly, I think that, at least the more impactful changes proposed, align with the project's [core principles](https://github.com/poliastro/poliastro/wiki/Core-principles), specifically `2` and `7`. If the tutorials could somehow become even more informative in an easy manner, I believe this is a win 🚀. It's a great practice for every FOSS project out there, too; specifically those that depend on long-term community engagement for their development and maintenance.

Let me know what you think and whether this is a sound idea/incoherent rambling/out of the project's scope @jorgepiloto @astrojuanlu

**Comment by Yash-10** [comment]
Hello @xlxs4, it's really true that there are small inconsistencies in some parts of the documentation and I totally agree with your thoughts. I have been making some small grammatical changes throughout the documentation locally, that I felt could be incorporated.

As you have suggested, other minor inconsistencies apart from some grammatical errors might also need further discussion.

So probably we could have some sort of discussion with @jorgepiloto @astrojuanlu to validate such changes.

Again, thanks for this feature request and it looks promising!

**Comment by jorgepiloto** [comment]
Thank you for this review on project's documentation, @xlxs4.

Let me start by discussing about the inconsistencies before doing so about tutorials. I share your opinion about having a uniform syntax in those documentation imports. The problem is that no tools are available (as far as I know) to format the code from the notebooks, as they are stored in `*.mystnb` files. Hence, we must select the appropriate syntax and try to identify all these inconsistencies by hard. Finally, when it comes to grammatical errors, those need to be fixed for sure, no discussion here.

Regarding the tutorials, it is true that we have a good collection of examples showing poliastro's main features and capabilities. However, they are not intended to be an introductory course to astrodynamics neither orbital mechanics and thus, no particular order is imposed among the different tutorials. I guess this might be a little beet confusing for newcomers... We could include some kind of "Explanations" section in our docs about basic concepts and how poliastro solves those. Issue #985 might also be added under this category. This last idea might be proposed for #1106...

While we wait for other users and contributors to share their ideas, let us try to solve for those inconsistencies you found 😉

**Comment by xlxs4** [comment]
> Let me start by discussing about the inconsistencies before doing so about tutorials. I share your opinion about having a uniform syntax in those documentation imports. The problem is that no tools are available (as far as I know) to format the code from the notebooks, as they are stored in `*.mystnb` files. Hence, we must select the appropriate syntax and try to identify all these inconsistencies by hard. Finally, when it comes to grammatical errors, those need to be fixed for sure, no discussion here.

Agreed, there should be a way to automatically run something (e.g. `black`) against the code in the mystnb cells, if someone wants to play around with that.

> Regarding the tutorials, it is true that we have a good collection of examples showing poliastro's main features and capabilities. However, they are not intended to be an introductory course to astrodynamics neither orbital mechanics and thus, no particular order is imposed among the different tutorials. I guess this might be a little beet confusing for newcomers... We could include some kind of "Explanations" section in our docs about basic concepts and how poliastro solves those. Issue #985 might also be added under this category. This last idea might be proposed for #1106...

The Explanations/Introduction/Footnote section is another great idea. For example, I saw an answer in an old issue referencing "Orbital Mechanics for Engineering Students" from Curtis as good introductory material, which is already linked [here](https://docs.poliastro.space/en/stable/references.html#books-and-papers). Things like that (aka pointers for beginners) could be put inside such a section.

> While we wait for other users and contributors to share their ideas, let us try to solve for those inconsistencies you found 😉

Sure! Let's leave this open, then 🚀

**Comment by xlxs4** [comment]
> Agreed, there should be a way to automatically run something (e.g. black) against the code in the mystnb cells, if someone wants to play around with that.

Using a [regexp](https://regex101.com/r/4NRt2q/1) to extract the code and run it against checkers, for an example.

**Comment by astrojuanlu** [comment]
> The problem is that no tools are available (as far as I know) to format the code from the notebooks, as they are stored in *.mystnb files. 

Look!

<pre><code>
(.venv) juanlu@voyager:~/Development/poliastro/poliastro/docs/source/examples$ jupytext --pipe black Plotting\ in\ 3D.mystnb 
[jupytext] Reading Plotting in 3D.mystnb in format mystnb
[jupytext] Executing black -
reformatted -
All done! ✨ 🍰 ✨
1 file reformatted.
[jupytext] Writing 'Plotting in 3D.mystnb' in format mystnb:myst (destination file replaced [use --update to preserve cell outputs and ids])
(.venv) juanlu@voyager:~/Development/poliastro/poliastro/docs/source/examples$ git diff
diff --git a/docs/source/examples/Plotting in 3D.mystnb b/docs/source/examples/Plotting in 3D.mystnb
index 1d227b51..de1c2994 100644
--- a/docs/source/examples/Plotting in 3D.mystnb        
+++ b/docs/source/examples/Plotting in 3D.mystnb        
@@ -3,8 +3,8 @@ jupytext:
   text_representation:
     extension: .mystnb
     format_name: myst
-    format_version: 0.12
-    jupytext_version: 1.9.1
+    format_version: 0.13
+    jupytext_version: 1.11.0
 kernelspec:
   display_name: Python 3
   language: python
@@ -23,6 +23,7 @@ from poliastro.plotting import *
 
 ```{code-cell}
 import plotly.io as pio
+
 pio.renderers.default = "notebook_connected"
 ```
 
@@ -62,9 +63,7 @@ from poliastro.util import time_range
 date_launch = time.Time("2011-11-26 15:02", scale="utc").tdb
 date_arrival = time.Time("2012-08-06 05:17", scale="utc").tdb
 
-earth = Ephem.from_body(
-    Earth, time_range(date_launch, end=date_arrival, periods=50)
-)
+earth = Ephem.from_body(Earth, time_range(date_launch, end=date_arrival, periods=50))
 ```
 
 ```{code-cell}
</pre></code>

**Comment by astrojuanlu** [comment]
> An objection to this, however could very well be that:

It's a very good observation @xlxs4. I propose we:

- Do some review of code and import inconsistencies, as you suggested
- Do some small typo and grammar correction,
- Leave large text changes to a technical writer when we apply to Google Season of Docs (hopefully next year)

@xlxs4 Do you want to take care about the first or two first points?

Also, congratulations on this excellent analysis of the repository! Loved the links to existing issues :)

**Comment by xlxs4** [comment]
Haha, I didn't know `jupytext` was a thing 😄 
Sure, can do @astrojuanlu! sounds good 🛰️

**Comment by astrojuanlu** [comment]
CLosed in #1133
